File: ui/qt/theme.cpp
#include <QtWidgets/QApplication>
#include "theme.h"

namespace Theme
{
	QColor background = QColor(44, 44, 44);
	QColor backgroundDark = QColor(34, 34, 34);
	QColor backgroundWindow = QColor(56, 56, 56);
	QColor backgroundHighlight = QColor(64, 64, 64);
	QColor content = QColor(224, 224, 224);
	QColor light = QColor(120, 120, 120);
	QColor disabled = QColor(160, 160, 160);
	QColor selection = QColor(96, 96, 96);
	QColor selectionLight = QColor(128, 128, 128);
	QColor green = QColor(162, 217, 175);
	QColor red = QColor(222, 143, 151);
	QColor blue = QColor(128, 198, 233);
	QColor cyan = QColor(142, 230, 237);
	QColor lightCyan = QColor(176, 221, 228);
	QColor orange = QColor(237, 189, 129);
	QColor yellow = QColor(237, 223, 179);
	QColor magenta = QColor(218, 196, 209);
}


QColor MixColor(const QColor& a, const QColor& b, uint8_t alpha)
{
	uint8_t red = (uint8_t)((((uint16_t)a.red() * (255 - alpha)) +
		((uint16_t)b.red() * alpha)) / 255);
	uint8_t green = (uint8_t)((((uint16_t)a.green() * (255 - alpha)) +
		((uint16_t)b.green() * alpha)) / 255);
	uint8_t blue = (uint8_t)((((uint16_t)a.blue() * (255 - alpha)) +
		((uint16_t)b.blue() * alpha)) / 255);
	return QColor(red, green, blue);
}


float relativeFontSize(float mult)
{
	return QApplication::font().pointSizeF() * mult;
}


QFont fontOfRelativeSize(float mult, int weight, bool italic)
{
	QFont result(QApplication::font());
	result.setWeight(weight);
	result.setItalic(italic);
	result.setPointSizeF(result.pointSizeF() * mult);
	return result;
}

File: src/theme.rs
use eframe::egui::Color32;

pub enum Theme {
    Background,
    BackgroundDark,
    BackgroundWindow,
    BackgroundHighlight,
    Content,
    Light,
    Disabled,
    Selection,
    SelectionLight,
    Green,
    Red,
    Blue,
    DarkBlue,
    Cyan,
    LightCyan,
    Orange,
    Yellow,
    Magenta,
}

impl Into<Color32> for Theme {
    fn into(self) -> Color32 {
        match self {
            Theme::Background => Color32::from_rgb(44, 44, 44),
            Theme::BackgroundDark => Color32::from_rgb(34, 34, 34),
            Theme::BackgroundWindow => Color32::from_rgb(56, 56, 56),
            Theme::BackgroundHighlight => Color32::from_rgb(64, 64, 64),
            Theme::Content => Color32::from_rgb(224, 224, 224),
            Theme::Light => Color32::from_rgb(120, 120, 120),
            Theme::Disabled => Color32::from_rgb(160, 160, 160),
            Theme::Selection => Color32::from_rgb(96, 96, 96),
            Theme::SelectionLight => Color32::from_rgb(128, 128, 128),
            Theme::Green => Color32::from_rgb(162, 217, 175),
            Theme::Red => Color32::from_rgb(222, 143, 151),
            Theme::Blue => Color32::from_rgb(128, 198, 233),
            Theme::DarkBlue => Color32::from_rgb(86, 121, 139),
            Theme::Cyan => Color32::from_rgb(142, 230, 237),
            Theme::LightCyan => Color32::from_rgb(176, 221, 228),
            Theme::Orange => Color32::from_rgb(237, 189, 129),
            Theme::Yellow => Color32::from_rgb(237, 223, 179),
            Theme::Magenta => Color32::from_rgb(218, 196, 209),
        }
    }
}
